### PR TITLE
Upgrade deployer application

### PR DESCRIPTION
**Pull request by IlonaShishov**
- expanded Application struct to include configmaps and secrets
- expanded deployer config to include cluster struct
- revised application customization capabilities
- renamed components, files and folders
- added simple deployer

### CODE REVIEW

**Review by dmartinol — APPROVED** [review]
Fine for me, do you want to merge them?

**Review by dmartinol — APPROVED** [review]

**Review by dmartinol — APPROVED** [review]

**Review by dmartinol — APPROVED** [review]
LGTM

File: replica-builder/deployer/main.go [hunk 1]
```diff
@@ -16,18 +17,29 @@ var (
...
+	// get deployer config yaml as input
 	if len(os.Args) != 2 {
 		log.Fatal("Expected 1 argument, got ", len(os.Args)-1)
 	}
 
-	// init ApplicationConfig object
-	applicationConfig := config.ReadApplicationConfig(os.Args[1])
-	pretty.Printf("Exporting application %# v", applicationConfig)
+	// Unmarshal deployer config and get cluster and application configs
+	ComponentConfig := config.ReadDeployerConfig(os.Args[1])
```
Comment (dmartinol) [inline replica-builder/deployer/main.go:26]: why capital C? I don't think this is public variable, right?

**Review by dmartinol — APPROVED** [review]
LGTM

File: replica-builder/deployer/main.go (hunk 1 above)
Comment (IlonaShishov) [reply replica-builder/deployer/main.go:26]: you are right, its a typo from copy/paste, its supposed to be small c, ill push a fix right now

**Review by dmartinol — APPROVED** [review]
LGTM

**Pull request merged**